snippets/curl/get_nb_vmservers.sh

```diff
@@ -0,0 +1,111 @@
+!/bin/sh
+
+#####################n#####################################################
+
+# This script demonstrates the usage of netbackup REST API for listing
+# the vmservers
+
+# This script requires jq command-line JSON parser
+# if your system does not have jq installed, this will not work
+# jq can be downloaded from here: https://github.com/stedolan/jq/releases
+
+###########################################################################
+
+port=1556
+master_server=""
+username=""
+password=""
+domainname=""
+domaintype=""
+
+showHelp()
+{
+        echo ""
+        echo "Invalid command parameters"
+        echo "Usage:"
+        echo "./get_nb_vmservers.sh -nbmaster <master_server> -username <username> -password <password> -domainname <dname> -domaintype <unixpwd/nt>"
+        echo ""
+        exit 1
+}
+
+parseArguments()
+{
+        if [ $# -lt 6 ]; then
+                showHelp
+        fi
+
+        while [ "$1" != "" ]; do
+                case $1 in
+                        -nbmaster)
+                                master_server=$2
+                                ;;
+                        -username)
+                                username=$2
+                                ;;
+                        -password)
+                                password=$2
+                                ;;
+                        -domainname)
+                                domainname=$2
+                                ;;
+                        -domaintype)
+                                domaintype=$2
+                                ;;
+                        *)
+                                showHelp
+                                ;;
+                esac
+                shift 2
+        done
+
+        if [ -z "$master_server" ] || [ -z "$username" ] || [ -z "$password" ] || [ -z "$domainname" ] || [ -z "$domaintype" ]; then
+                showhelp
+        fi
+
+        if [ "${domaintype^^}" = "WINDOWS" ] || [ "${domaintype^^}" = "NT" ]; then
+                domaintype="nt"
+        fi
+}
+
+uriencode()
+{
+        jq -nr --arg v "$1" '$v|@uri';
+}
+###############main############
+
+parseArguments "$@"
+
+master_server=$master_server.$domainname
+basepath="https://$master_server:$port/netbackup"
+content_header='content-type:application/json'
+
+##############login#############
+
+uri="$basepath/login"
+echo $uri
+
+data=$(jq --arg name $username --arg pass $password --arg dname $domainname --arg dtype $domaintype \
```

Comment:
Quote shell parameter expansion to avoid syntax errors form unexpected contents. Especially for the password, which might contain anything.

```suggestion
data=$(jq --arg name "$username" --arg pass "$password" --arg dname "$domainname" --arg dtype "$domaintype" \
```

`dname` and `dtype` don't get used in the jq program.